src/Type/Php/RegexNonCapturingGroup.php

```diff
@@ -0,0 +1,29 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Type\Php;
+
+class RegexNonCapturingGroup
```

Comment:
the optional groups detection, now also takes non-capturing groups into account... this do not capture results itself, but might impact nested capturing groups